homeassistant/components/command_line/__init__.py

```diff
@@ -2,49 +2,77 @@
 from __future__ import annotations
 
 import logging
-import subprocess
+from typing import Any
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PLATFORM, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def call_shell_with_timeout(
-    command: str, timeout: int, *, log_return_code: bool = True
-) -> int:
-    """Run a shell command with a timeout.
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Command Line component to import config."""
 
-    If log_return_code is set to False, it will not print an error if a non-zero
-    return code is returned.
-    """
-    try:
-        _LOGGER.debug("Running command: %s", command)
-        subprocess.check_output(
-            command, shell=True, timeout=timeout  # nosec # shell by design
-        )
-        return 0
-    except subprocess.CalledProcessError as proc_exception:
-        if log_return_code:
-            _LOGGER.error("Command failed: %s", command)
-        return proc_exception.returncode
-    except subprocess.TimeoutExpired:
-        _LOGGER.error("Timeout for command: %s", command)
-        return -1
-    except subprocess.SubprocessError:
-        _LOGGER.error("Error trying to exec command: %s", command)
-        return -1
-
-
-def check_output_or_log(command: str, timeout: int) -> str | None:
-    """Run a shell command with a timeout and return the output."""
-    try:
-        return_value = subprocess.check_output(
-            command, shell=True, timeout=timeout  # nosec # shell by design
+    if Platform.NOTIFY not in config:
+        return True
+
+    entry: dict[str, Any]
+    for entry in config[Platform.NOTIFY]:
+        if entry[CONF_PLATFORM] == DOMAIN:
+            _LOGGER.warning(
+                # Command Line config flow added in 2022.6 and should be removed in 2022.8
+                "Configuration of the Command Line Notify platform in YAML is deprecated and "
+                "will be removed in Home Assistant 2022.8; Your existing configuration "
+                "has been imported into the UI automatically and can be safely removed "
+                "from your configuration.yaml file"
+            )
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_IMPORT},
+                    data={**entry, CONF_PLATFORM: "notify"},
+                )
+            )
```

Comment:
This integration does not connect to a device or service, therefore, ADR-0010 does not apply here. As this is an integration that defines its own protocol.

For that reason I expected it to support both YAML and UI  (and no import happens).

Reply:
That's fine. I will change accordingly.

Reply:
Yaml is now returned and also removed the breaking change as there is none